azure/build-and-deploy-azure.sh

```diff
@@ -0,0 +1,174 @@
+
+helm list --namespace kube-system | grep 'aad-pod-identity' &>/dev/null
+if [ $? == 0 ]; then
+  echo "Azure pod identity chart already installed"
+else
+  helm upgrade --install aad-pod-identity aad-pod-identity/aad-pod-identity #NO LONGER WORKS BECAUSE OF OUR CONFIUGRATION (RESTRICTED IN DEFAULT)
+fi
+
+#END TO BE REPLACED WITH https://azure.github.io/azure-workload-identity/docs/installation.html
+
+
+# Preparing calico via Helm
+echo "preparing calico via Helm"
+helm repo add projectcalico https://docs.projectcalico.org/charts
+helm upgrade --install calico projectcalico/tigera-operator
+
+
+echo "Generate secret manager challenge secret 2"
+az keyvault secret set --name wrongsecret-2 --vault-name "${AZ_KEY_VAULT_NAME}" --value "$(openssl rand -base64 16)" >/dev/null
+
+echo "Generate secret manager challenge secret 3"
+az keyvault secret set --name wrongsecret-3 --vault-name "${AZ_KEY_VAULT_NAME}" --value "$(openssl rand -base64 16)" >/dev/null
+
+echo "Fill-out the secret volume manifest template"
+envsubst <./k8s/secret-volume.yml.tpl >./k8s/secret-volume.yml
+
+echo "Apply secretsmanager storage volume"
+kubectl apply -f./k8s/secret-volume.yml
+
+envsubst <./k8s/pod-id.yml.tpl >./k8s/pod-id.yml
+envsubst <./k8s/secret-challenge-vault-deployment.yml.tpl >./k8s/secret-challenge-vault-deployment.yml
+
+kubectl apply -f./k8s/pod-id.yml
+
+while [[ $(kubectl --namespace=default get pods -l "app.kubernetes.io/component=mic" -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True True" ]]; do echo "waiting for component=mic" && sleep 2; done
+while [[ $(kubectl --namespace=default get pods -l "app.kubernetes.io/component=nmi" -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True True True" ]]; do echo "waiting for component=nmi" && sleep 2; done
+
+
+echo "Installing metrics api-server"
+kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml
+
+wait
+
+# if passed as arguments, use those
+# otherwise, create new default values
+
+if [[ -z $APP_PASSWORD ]]; then
+  echo "No app password passed, creating a new one"
+  APP_PASSWORD="$( uuidgen | sed 's/[-]//g')"
+else
+  echo "App password already set"
+fi
+
+if [[ -z $CREATE_TEAM_HMAC ]]; then
+  CREATE_TEAM_HMAC="$(openssl rand -base64 24)"
+else
+  echo "Create team HMAC already set"
+fi
+
+if [[ -z $COOKIE_PARSER_SECRET ]]; then
+  COOKIE_PARSER_SECRET="$(openssl rand -base64 24)"
+else
+  echo "Cookie parser secret already set"
+fi
+
+echo "App password is ${APP_PASSWORD}"
+helm upgrade --install mj ../helm/wrongsecrets-ctf-party \
+  --set="balancer.env.K8S_ENV=azure" \
+  --set="balancer.env.REACT_APP_ACCESS_PASSWORD=${APP_PASSWORD}" \
+  --set="balancer.env.REACT_APP_CREATE_TEAM_HMAC_KEY=${CREATE_TEAM_HMAC}" \
+  --set="balancer.cookie.cookieParserSecret=${COOKIE_PARSER_SECRET}"
+
+  # TODO: Support azure storage account instead
+  # --set="balancer.env.REACT_APP_S3_BUCKET_URL=s3://${STATE_BUCKET}" \
```

Comment:
The app currently shows:
```
Welcome to the WrongSecrets CTF Party! This CTF uses the following domains:
This domain: here is where you can do your exercises
The domain where you can have to submit your answers in exchange for answers that are worth points: https://wrongsecrets-ctf.herokuapp.com/
The domain where you provide your answer found in exchange for points: https://ctfd.io/
Optionally: the storage bucket with Terraform state for the cloud challenges: s3://. For this you will need credentials that will be provided to you as part of the CTF instructions.
```
this will need some updating to be contextual aware of what the actual urls would be.

Reply:
I fixed the storage bucket part and added a section in the Readme file how to configure the domains.